src/components/ButtonModal.tsx

```diff
@@ -0,0 +1,36 @@
+import { ReactNode, useRef } from "react";
+import { useMenuTrigger } from "react-aria";
+import { useMenuTriggerState } from "react-stately";
+import {
+  isIconButton,
+  isTextButton,
+  OverlayTrigger,
+  OverlayTriggerProps,
+} from "src/components/internal/OverlayTrigger";
+import { useTestIds } from "src/utils";
+import { ContextualModal } from "./internal/ContextualModal";
+
+export interface ContextualModalProps
```

Comment:
Suggestion: Probably rename this to `ButtonModalProps` so that it matches the component name?

Reply:
yes, will do.